sharded_queue/api.lua

```diff
@@ -92,6 +106,10 @@ function sharded_tube.take(self, timeout, options)
     local wait_part = 0.01 -- maximum waiting time in second
     local wait_max = options.wait_max or self.wait_max or time.MAX_TIMEOUT
 
+    if not validate_wait_max(wait_max) then
```

Comment:
I think we should call `validate_options` for `options` and unify `wait_factor` assignment with this